mock-etcd/src/mock_etcd.rs

```diff
@@ -776,11 +792,12 @@ impl Lock for MockEtcd {
         let inner_clone = Arc::<Mutex<MockEtcdInner>>::clone(&self.inner);
         let task = async move {
             loop {
-                log::debug!("inner lock `lock`");
                 let mut inner = inner_clone.lock().await;
+                log::debug!("inner lock `lock`");
+
                 if inner.lock_map.contains(req.get_name()) {
-                    Timer::after(Duration::from_secs(1)).await;
                     drop(inner);
+                    Timer::after(Duration::from_secs(1)).await;
```

Comment:
What's the purpose of this waiting?

Reply:
This is the original code that already exists. It uses a polling method to determine if the lock is still present, and delaying can reduce the frequency of acquiring the lock.